### PR TITLE
Enhance the spellcheck workflow

### PR DESCRIPTION
**Pull request by gdha**
* Type:  **Other?**

* Impact: **Low**

* Description of the changes in this pull request: After merging PR https://github.com/rear/rear/pull/3269 we wanted to enhance the spelling check workflow to use codespell-project/actions-codespell instead of building our own image. Furthermore, now we only have to check the added or changed files and not the complete master branch.

### CODE REVIEW

**Comment by gdha** [comment]
Yep - it works fine as it found a spelling error in its own workflow ;-)
```
Error: ./.github/workflows/spellcheck.yml:22: seperate ==> separate
Codespell found one or more problems
```

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
+        run: echo ${{ env.CHANGED_FILES }}
+          
+      - name: 🔸 Check for Spelling Errors for Changed Files
+        uses: codespell-project/actions-codespell@v2
+        with:
+          check_filenames: true
+          check_hidden: true
+          ignore_words_file: .codespellignore
+          skip: nothing
+          path: ${{ env.CHANGED_FILES }}
```
Comment (schlomo) [inline .github/workflows/spellcheck.yml:36]: @gdha do you see any chance to harden that against paths/files with spaces or stupid chars in them?

Comment (gdha) [reply .github/workflows/spellcheck.yml:36]: Let me check on this

**Comment by schlomo** [comment]
@gdha I'm wondering if another approach might be simpler: Use the PR files changed info to create an exclude file that lists all the files that *didn't* change and then let codespell work on the full repo? You can simply append the not-changed files to the potentially existing exclude list file and that should be enough. And safe with regard to funny file names.

**Comment by gdha** [comment]
@schlomo it seems to work as you suggested now - see https://github.com/rear/rear/actions/runs/9907387431/job/27370960104

**Review by schlomo — APPROVED** [review]

**Comment by schlomo** [comment]
Feel free to merge

**Pull request merged**